docs/data/sql_functions.yml

```
@@ -516,7 +516,14 @@ conditional:
   - sql: NULLIF(value1, value2)
     description: Returns NULL if value1 is equal to value2; returns value1 otherwise. E.g., NULLIF(5, 5) returns NULL; NULLIF(5, 0) returns 5.
   - sql: COALESCE(value1, value2 [, value3]*)
-    description: Returns the first value that is not NULL from value1, value2, .... E.g., COALESCE(NULL, 5) returns 5.
+    description: |
+      Coalesce specifies a series of expressions, and returns the first expression whose value is not null.
+
+      If all the expressions evaluate as null, coalesce returns a null value.
+      The return type of the coalesce function is the least common type among its arguments, and it's nullable if all its arguments are nullable.
```

The function itself doesn't specify any expressions, but rather the user passes them in, so I would rephrase this slightly. Since this is the SQL docs, we should use NULL for consistency. Also, "least common type" sounds a bit odd to me, it kind of sounds like "we'll count the types of the operands and whichever one appears least often will be used".

```suggestion
      Returns the first argument that is not NULL.
      
      If all arguments are NULL, it returns NULL as well. The return type is the least restrictive, common type of all of its arguments. The return type is nullable if all arguments are nullable as well.
```

```suggestion
      Coalesce specifies a series of expressions, and returns the first expression whose value is not null.

      If all the expressions evaluate as null, coalesce returns a null value.
      The return type of the coalesce function is the least common type among its arguments, and it's nullable if all its arguments are nullable.
```

---

Of course this goes for all places of the docs.